key.js

```diff
@@ -1,18 +1,321 @@
+            gameState.gameAudio.getClickAudio(),
+            gameState.gameAudio.getKeyboardSounds(),
+            keyboardSounds.getInvalidAudio(),
+            keyboardSounds.getGreenAudio(),
+            keyboardSounds.getRedAudio()
+        ];
+        sounds.forEach(audio => audio.pause());
+    }
+    function unMuteAudio() {
+        gameState.sound = true;
+        const sounds = [
+            gameState.gameAudio.getGameAudio(),
+            gameState.gameAudio.getClickAudio(),
+            gameState.gameAudio.getKeyboardSounds(),
+            keyboardSounds.getInvalidAudio(),
+            keyboardSounds.getGreenAudio(),
+            keyboardSounds.getRedAudio()
+        ];
+        sounds.forEach(audio => {
+            audio.currentTime = 0;
+            audio.play()
+        });
+    }
+    function decrementGameLife() {
+        gameState.lives--;
+    }
+
+    function printGameState() {
+        console.log(gameState);
+    }
+
+    return {
+        startGame,
+        gameOver,
+        levelUp,
+        resetGameLivesAndTimer,
+        resetGameState,
+        muteAudio,
+        unMuteAudio,
+        decrementGameLife,
+        printGameState,
+        getGameLife,
+        endGame,
+        gameMusicPlay,
+        gameMusicStop
+    }
+    // I. create a game state object (ensure it's a deep copy of initialGameState)
+    // II. create utility methods to manipulate the game state object
+    // 1. incrementGameLevel
+    // 2. getGameLevel
+    // 3. startGame
+    // 4. endGame
+    // 5. isGameRunning
+    // 6. isGameOver
+    // 7. getGameLife
+    // 8. decrementGameLife
+    // 9. resetGameState
+    // 10. getGameAudio
+    // 11. printGameState // just for debugging
+    // 12. <create any other method you feel that makes sense>
+    // III. return an object with all the aforementioned utility methods from 1..9.
+
+    // return {
+    //     incrementGameLevel: () => gameState.levelsInfo.currentLevel++
+    // }
+})();
+// IV. Refactor the code below to use the gameModule above for creating the game logic.
 // Button Animation Logic :
 
+const viewModule = (function () {
+    const DOMStrings = {
+        entireContainer: 'container',
+        changeHeading: 'main_heading',
+        defaultLives: 'default',
+        currentLives: 'lives',
+        mainButton: 'main',
+        liveDisplay: 'heart',
+        timerDisplay: 'timer',
+        keyDirections: 'btns',
+        randomKeyArrows: 'arrows',
+        keyboardContents: 'contents'
+    }
+    const newSequence = () => {
+        const arr = [];
+        let i = 10;
+        while (i--) {
+            const random = Math.random();
+            if (random <= .25) {
+                arr.push("&uarr;");
+            }
+            else if (random <= .50) {
+                arr.push("&rarr;");
+            }
+            else if (random <= .75) {
+                arr.push("&rarr;");
+            }
+            else {
+                arr.push("&darr;");
+            }
```

Comment:
Shouldn't there be an `&larr;`?  You're only generating ↑, ↓, and → arrows, where's the ← arrow?

```suggestion
            if (random <= .25) {
                arr.push("&uarr;");
            }
            else if (random <= .50) {
                arr.push("&rarr;"); // Change this to `&larr;`
            }
            else if (random <= .75) {
                arr.push("&rarr;"); // or this to `&larr;`
            }
            else {
                arr.push("&darr;");
            }
```